src/Parser/Php/Visitor/ForbiddenStaticMethodCallsVisitor.php

```diff
@@ -37,11 +37,11 @@ public function configure(array $options): void
      */
     public function leaveNode(Node $node): void
     {
-        if (!$node instanceof Node\Expr\StaticCall) {
+        if (!$node instanceof Node\Expr\StaticCall || !$node->class instanceof Node\Name) {
             return;
         }
 
-        $class = implode('\\', $node->class->parts);
+        $class = implode('\\', $node->class->getParts());
```

Comment:
Thanks for looking into this as well!

This one seems to be the party pooper:
https://github.com/nikic/PHP-Parser/releases/tag/v4.16.0

Maybe a `method_exists` check could temper psalm on older versions of php-parser?

Reply:
wouldn't it be better solution to update the version of nikic/php-parser in composer instead of making exceptions in code?

from "nikic/php-parser": "~4.13" to "nikic/php-parser": "^4.16"

Reply:
The problem is that the php-parser currently is a dev dependency (as it is opt-inable as task) and that by just bumping it, it might cause exceptions for users that are using an older version of php-parser.

Ideally, we should either extract the parser task into either a new extension (with it's own version requirements) or deprecate it completely.
Nowadays with psalm and phpstan, writing an additional extension for that is probably a better way then writing your own phpparser visitors. The visitors we provide can already be achieved by those package iirc.